controllers/memcached/memcached_controller.go

```diff
@@ -199,6 +249,27 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 	}
 	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
 
+	//
+	// create hash over all the different input resources to identify if any those changed
+	// and a restart/recreate is required.
+	//
+	hashOfHashes, err := util.HashOfInputHashes(inputHashEnv)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if hashMap, changed := util.SetHash(instance.Status.Hash, common.InputHashName, hashOfHashes); changed {
+		// Hash changed and instance status should be updated (which will be done by main defer func),
+		// so update all the input hashes and return to reconcile again
+		instance.Status.Hash = hashMap
+		for k, s := range inputHashEnv {
+			var envVar corev1.EnvVar
+			s(&envVar)
+			instance.Status.Hash[k] = envVar.Value
+		}
+		util.LogForObject(helper, fmt.Sprintf("Input hash changed %s", hashOfHashes), instance)
```

Comment:
we should stop using `util.LogForObject` and instead use the `Log.Info()`. same for redis https://github.com/openstack-k8s-operators/infra-operator/blob/main/controllers/redis/redis_controller.go#L263

Reply:
Ack, will do the other controllers as well (redis and galera)